Brokerages/GDAX/GDAXBrokerage.Messaging.cs

```diff
@@ -226,19 +229,32 @@ private void OrderMatch(string data)
                 direction = message.Side == "sell" ? OrderDirection.Sell : OrderDirection.Buy;
             }
 
+            decimal totalOrderFee;
+            if (!_orderFees.TryGetValue(orderId, out totalOrderFee))
+            {
+                totalOrderFee = GetFee(order);
+                _orderFees[orderId] = totalOrderFee;
+            }
+
+            // apply order fee on a pro rata basis
+            var orderFee = totalOrderFee * Math.Abs(message.Size) / Math.Abs(order.Quantity);
+
             var orderEvent = new OrderEvent
             (
-                cached.First().Key, symbol, message.Time, status,
+                orderId, symbol, message.Time, status,
                 direction,
                 message.Price, direction == OrderDirection.Sell ? -message.Size : message.Size,
-                GetFee(cached.First().Value), $"GDAX Match Event {direction}"
+                orderFee, $"GDAX Match Event {direction}"
             );
 
             //if we're filled we won't wait for done event
             if (orderEvent.Status == OrderStatus.Filled)
             {
-                Orders.Order outOrder = null;
-                CachedOrderIDs.TryRemove(cached.First().Key, out outOrder);
+                Order outOrder;
```

Comment:
Perhaps this conditional should read `if (orderEvent.Status.IsClosed())`

I'm not sure if canceled events get pumped through here as well, but we would certainly want to clean `CachedOrderIDs` and `_orderFees` each time an order transitions into the closed stated, not just the filled state.